### ISSUE TITLE
Windows bug in 'pipx uninstall'

### ISSUE THREAD

**Issue by OdinTech3** [pipxproject/pipx#210]
I think I encountered a bug while uninstalling `jupyter` using `pipx uninstall <package>`

```bash
{ ~ } » pipx uninstall jupyter 

Traceback (most recent call last):
  File "c:\users\nickmuoh\scoop\apps\python\3.7.4\lib\runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "c:\users\nickmuoh\scoop\apps\python\3.7.4\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\nickmuoh\Scoop\apps\python\current\Scripts\pipx.exe\__main__.py", line 9, in <module>
  File "c:\users\nickmuoh\scoop\apps\python\3.7.4\lib\site-packages\pipx\main.py", line 544, in cli
    exit(run_pipx_command(parsed_pipx_args, binary_args))
  File "c:\users\nickmuoh\scoop\apps\python\3.7.4\lib\site-packages\pipx\main.py", line 201, in run_pipx_command
    commands.uninstall(venv_dir, package, LOCAL_BIN_DIR, verbose)
  File "c:\users\nickmuoh\scoop\apps\python\3.7.4\lib\site-packages\pipx\commands.py", line 449, in uninstall
    file.unlink()
  File "c:\users\nickmuoh\scoop\apps\python\3.7.4\lib\pathlib.py", line 1294, in unlink
    self._accessor.unlink(self)
PermissionError: [WinError 5] Access is denied: 'C:\\Users\\nickmuoh\\.local\\bin\\jupyter-kernel.exe'
```

Then when I try to run the command again I get this stack trace but with this Error:

`Traceback (most recent call last):                                                                                                                  File "c:\users\nickmuoh\scoop\apps\python\3.7.4\lib\runpy.py", line 193, in _run_module_as_main                                                     "__main__", mod_spec)                                                                                                                           File "c:\users\nickmuoh\scoop\apps\python\3.7.4\lib\runpy.py", line 85, in _run_code                                                                exec(code, run_globals)                                                                                                                         File "C:\Users\nickmuoh\Scoop\apps\python\current\Scripts\pipx.exe\__main__.py", line 9, in <module>                                              File "c:\users\nickmuoh\scoop\apps\python\3.7.4\lib\site-packages\pipx\main.py", line 544, in cli                                                   exit(run_pipx_command(parsed_pipx_args, binary_args))                                                                                           File "c:\users\nickmuoh\scoop\apps\python\3.7.4\lib\site-packages\pipx\main.py", line 201, in run_pipx_command                                      commands.uninstall(venv_dir, package, LOCAL_BIN_DIR, verbose)                                                                                   File "c:\users\nickmuoh\scoop\apps\python\3.7.4\lib\site-packages\pipx\commands.py", line 449, in uninstall                                         file.unlink()                                                                                                                                     self._accessor.unlink(self) 
FileNotFoundError: [WinError 2] The system cannot find the file specified: 'C:\\Users\\nickmuoh\\.local\\bin\\jupyter-nbextension.exe'`

Then I try to run it again and I get the same stack trace but with this error:
`FileNotFoundError: [WinError 2] The system cannot find the file specified: 'C:\\Users\\nickmuoh\\.local\\bin\\jupyter-notebook.exe'`

Try to run it again, then ....

`FileNotFoundError: [WinError 2] The system cannot find the file specified: 'C:\\Users\\nickmuoh\\.local\\bin\\jupyter-serverextension.exe'`

Then finally I got this:

`uninstalled jupyter! `

Any ideas about  what might be causing this

**Comment by cs01** [comment]
Thanks for reporting this. Could you recreate this, but with the `--verbose` flag passed to all commands, and paste the output here?

**Comment by sahensley** [comment]
Using `pipx` to install/uninstall Jupyter in WSL works, as long as `--install-deps` is passed, otherwise no packages are found.  I was able to replicate this issue and from what I can tell, it appears to be Windows specific.

The files listed in the "Overwriting" section of the log are the ones which produce the error message upon deletion during `pipx uninstall jupyter`.  As noted above by OdinTech3, uninstall can be run until all of those files are removed, then the next uninstall command will delete the jupyter venv.

<details>
 <summary>Windows pipx install --include-deps jupyter</summary>

```log
❯ py -m pipx install --include-deps jupyter
WARNING: You are using pip version 19.2.3, however version 19.3.1 is available.                                        
You should consider upgrading via the 'python -m pip install --upgrade pip' command.
  Overwriting file C:\Users\zzz\.local\bin\iptest.exe with C:\Users\zzz\.local\pipx\venvs\jupyter\Scripts\iptest.exe     
  Overwriting file C:\Users\zzz\.local\bin\iptest3.exe with C:\Users\zzz\.local\pipx\venvs\jupyter\Scripts\iptest3.exe   
  Overwriting file C:\Users\zzz\.local\bin\ipython.exe with C:\Users\zzz\.local\pipx\venvs\jupyter\Scripts\ipython.exe   
  Overwriting file C:\Users\zzz\.local\bin\ipython3.exe with C:\Users\zzz\.local\pipx\venvs\jupyter\Scripts\ipython3.exe                                                                                                                        
  Overwriting file C:\Users\zzz\.local\bin\jupyter-bundlerextension.exe with C:\Users\zzz\.local\pipx\venvs\jupyter\Scripts\jupyter-bundlerextension.exe                                                                                        
  Overwriting file C:\Users\zzz\.local\bin\jupyter-nbextension.exe with C:\Users\zzz\.local\pipx\venvs\jupyter\Scripts\  
  Overwriting file C:\Users\zzz\.local\bin\jupyter-notebook.exe with C:\Users\zzz\.local\pipx\venvs\jupyter\Scripts\jupyter-notebook.exe                                                                                                        
  Overwriting file C:\Users\zzz\.local\bin\jupyter-serverextension.exe with C:\Users\zzz\.local\pipx\venvs\jupyter\Scripts\jupyter-serverextension.exe                                                                                          
  installed package jupyter 1.0.0, Python 3.7.4                                                                          
  These apps are now globally available                                                                                   
   - iptest.exe                                                                                                           
   - iptest3.exe                                                                                                          
   - ipython.exe                                                                                                          
   - ipython3.exe                                                                                                         
   - jupyter-bundlerextension.exe                                                                                         
   - jupyter-console.exe                                                                                                  
   - jupyter-kernel.exe                                                                                                   
   - jupyter-kernelspec.exe                                                                                               
   - jupyter-nbconvert.exe                                                                                                
   - jupyter-nbextension.exe                                                                                              
   - jupyter-notebook.exe                                                                                                 
   - jupyter-qtconsole.exe                                                                                                
   - jupyter-run.exe                                                                                                      
   - jupyter-serverextension.exe                                                                                          
   - jupyter-trust.exe                                                                                                
   done!                                                                                                                  
```
</details>

<details>
 <summary>Windows pipx uninstall jupyter --verbose</summary>

```log

 py -m pipx uninstall jupyter --verbose
pipx > (run_pipx_command:134): Virtual Environment location is C:\Users\zzz\.local\pipx\venvs\jupyter
Traceback (most recent call last):
  File "C:\Users\zzz\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 193, in _run_module_as_main                
    "__main__", mod_spec)                                                                                                
  File "C:\Users\zzz\AppData\Local\Programs\Python\Python37\lib\runpy.py", line 85, in _run_code                           
    exec(code, run_globals)                                                                                              
  File "C:\Users\zzz\AppData\Local\Programs\Python\Python37\lib\site-packages\pipx\__main__.py", line 5, in <module>
         cli()                                                                                                                
  File "C:\Users\zzz\AppData\Local\Programs\Python\Python37\lib\site-packages\pipx\main.py", line 547, in cli              
    exit(run_pipx_command(parsed_pipx_args))                                                                             
  File "C:\Users\zzz\AppData\Local\Programs\Python\Python37\lib\site-packages\pipx\main.py", line 200, in run_pipx_command                                                                                                                        
    commands.uninstall(venv_dir, package, LOCAL_BIN_DIR, verbose)                                                        
  File "C:\Users\zzz\AppData\Local\Programs\Python\Python37\lib\site-packages\pipx\commands.py", line 458, in uninstall    
    file.unlink()                                                                                                        
  File "C:\Users\zzz\AppData\Local\Programs\Python\Python37\lib\pathlib.py", line 1294, in unlink                          
    self._accessor.unlink(self)                                                                                        
  FileNotFoundError: [WinError 2] The system cannot find the file specified: 'C:\\Users\\zzz\\.local\\bin\\iptest.exe' 
```

</details>

**Comment by dmwyatt** [comment]
I'm not sure if this is the same issue, but...probably?

```
> pipx upgrade poetry
  Overwriting file C:\Users\dusti\.local\bin\poetry.exe with C:\Users\dusti\.local\pipx\venvs\poetry\Scripts\poetry.exe
Traceback (most recent call last):
  File "c:\users\dusti\.pyenv\pyenv-win\versions\3.8.0-amd64\lib\runpy.py", line 192, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "c:\users\dusti\.pyenv\pyenv-win\versions\3.8.0-amd64\lib\runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "C:\Users\dusti\AppData\Roaming\Python\Python38\Scripts\pipx.exe\__main__.py", line 9, in <module>
  File "C:\Users\dusti\AppData\Roaming\Python\Python38\site-packages\pipx\main.py", line 547, in cli
    exit(run_pipx_command(parsed_pipx_args))
  File "C:\Users\dusti\AppData\Roaming\Python\Python38\site-packages\pipx\main.py", line 187, in run_pipx_command
    commands.upgrade(
  File "C:\Users\dusti\AppData\Roaming\Python\Python38\site-packages\pipx\commands.py", line 223, in upgrade
    _expose_apps_globally(LOCAL_BIN_DIR, metadata.app_paths, package, force=force)
  File "C:\Users\dusti\AppData\Roaming\Python\Python38\site-packages\pipx\commands.py", line 512, in _expose_apps_globally
    _copy_package_apps(local_bin_dir, app_paths, package)
  File "C:\Users\dusti\AppData\Roaming\Python\Python38\site-packages\pipx\commands.py", line 526, in _copy_package_apps
    dest.unlink()
  File "c:\users\dusti\.pyenv\pyenv-win\versions\3.8.0-amd64\lib\pathlib.py", line 1303, in unlink
    self._accessor.unlink(self)
PermissionError: [WinError 5] Access is denied: 'C:\\Users\\dusti\\.local\\bin\\poetry.exe'
```

**Comment by itsayellow** [comment]
Fixed with #541
